server/src/main/java/org/elasticsearch/common/settings/FallbackSecureSettings.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.settings;
+
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.common.hash.MessageDigests;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * An implementation of secure settings that fall back to the YML settings.
+ *
+ * WARNING: this is a temporary class to be used only in the context of Stateless. It applies only to YML settings with a predetermined
+ * prefix. Other secure settings do not fall back to YML settings.
+ */
+public class FallbackSecureSettings implements SecureSettings {
```

Comment:
This isn't really a "fallback" since the keystore will always be empty. I would instead name this something like "StatelessSecureSettings".

Reply:
Are you running this locally? Was the keystore a fresh file, or did you run it once and *then* try disabling security? This looks like auto security was setup. Can you please try ensuring there is no keystore, and that security is disabled (which should skip auto security config), and then if you still get the above error open an issue? This shouldn't be necessary, and in the future serverless setup we need to ensure the system will operate without the keystore.

Reply:
Oh yes I did run it once so it created the auto security config. I ran it from scratch with the following added settings

```
xpack.security.enabled: false
xpack.security.enrollment.enabled: false
xpack.security.http.ssl.enabled: false
xpack.security.transport.ssl.enabled: false
```

and now it does not do the auto security config and it does not show the error I mentioned.

Since it seems to work like that, I amended the PR to remove the SecuritySettings I kept behind the scenes, so now in Stateless it uses wholly the StatelessSecuritySettings without the keystore. This also simplified the code and the new tests.